### PR TITLE
Create topology graph component

### PR DESCRIPTION
**Pull request by dgutride**
This ports the kubernetes-ui component created by Red Hat over to PatternFly with new functionality that has been used by other products.

Inclues unit tests, less for styling and an example of how to style specific nodes, search and filter.

### CODE REVIEW

File: src/charts/topology/examples/topology-view.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+ * @description
+ *   Component for rendering a topology chart.  Individual nodes and relationships can be represented with this view.  CSS is especially important for rendering the noes and lines.  The example inline contains specific examples that can be used to change the icon size and the line type of the relationships.
+ *
+ *   In addition; searching, filtering and label visibility is also supported.<br/>
+ *
+ * @param {object} items items to display in the topology chart, each is represented as an individual node.  The keys of this object are used in the relations attribute. The items should have a item.kind attribute, as well as the usual item.metadata and so on.:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.name    - name of the item the node represents
+ * <li>.status  - optional status of the node (can be used to differentiate the circle color)
+ * <li>.kind    - the kind of node
```
Comment (dtaylor113) [inline src/charts/topology/examples/topology-view.js]: Can you elaborate on what 'kinds' of charts are supported, and what the 'kind' values may be?
Thanks,
- Dave

File: src/charts/topology/topology.component.js [hunk 2]
```diff
@@ -0,0 +1,630 @@
+angular.module('patternfly.charts').component('pfTopology', {
+  bindings: {
+    items: '=',
+    relations: '=',
```
Comment (dtaylor113) [inline src/charts/topology/topology.component.js]: Hi, do these have to be two way bindings?

**Comment by dtaylor113** [comment]
LGTM :+1:

File: src/charts/charts.less [hunk 3]
```diff
@@ -186,3 +186,121 @@ pf-c3-chart {
...
+}
+
+.pf-topology-svg g circle.unknown {
+  stroke: #bbb;
+}
+
+.pf-topology-svg g text.glyph {
+  font-size: 20px;
+  fill: #1186C1;
+}
```
Comment (jeff-phillips-18) [inline src/charts/charts.less]: Lots of additions to A-PF, these should be in Patternfly. Is there a backlog story to move this to Patternfly?

File: src/charts/topology/examples/topology-view.js [hunk 4]
```diff
@@ -0,0 +1,314 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.charts.component:pfTopology
+ * @restrict E
+ *
+ * @description
+ *   Component for rendering a topology chart.  Individual nodes and relationships can be represented with this view.  CSS is especially important for rendering the noes and lines.  The example inline contains specific examples that can be used to change the icon size and the line type of the relationships.
```
Comment (jeff-phillips-18) [inline src/charts/topology/examples/topology-view.js]: Typo: noes -> nodes

File: src/charts/topology/examples/topology-view.js [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+          "icon": "",
+          "fontfamily": "PatternFlyIcons-webfont"
+        },
+        "ContainerManager": {
+          "type": "glyph",
+          "icon": "",
+          "fontfamily": "PatternFlyIcons-webfont"
+        }
+			},
+    });
```
Comment (jeff-phillips-18) [inline src/charts/topology/examples/topology-view.js]: Funky indentation.

File: src/charts/topology/examples/topology-view.js [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+   .pf-topology-svg g.ContainerNode text.glyph {
+    font-size: 18px;
+  }
+
+   .pf-topology-svg g.ContainerManager text.glyph {
+    font-size: 18px;
+  }
+ </file>
+
```
Comment (jeff-phillips-18) [inline src/charts/topology/examples/topology-view.js]: Is it the expectation that applications will want to define each of these?

File: src/charts/topology/topology.component.js [hunk 7]
```diff
@@ -0,0 +1,626 @@
...
+      previousRelations,
+      previousKinds,
+      contextMenuShowing,
+      vs;
+    var cache = { };
+
+    ctrl.$onInit = function () {
+      $element.css("display", "block");
+      options = {"force": ctrl.force, "radius": ctrl.radius};
+      //graph = topologyGraph($element[0], notify, options);
```
Comment (jeff-phillips-18) [inline src/charts/topology/topology.component.js]: is this left in for a reason?

File: src/charts/topology/topology.component.js [hunk 8]
```diff
@@ -0,0 +1,626 @@
...
+      case 'Warning':
+      case 'Waiting':
+      case 'Pending':
+        return "warning";
+      case 'Unknown':
+      case 'Terminated':
+        return "unknown";
+      }
+    }
+  }
```
Comment (jeff-phillips-18) [inline src/charts/topology/topology.component.js]: May want to use case insensitive checks here.

File: src/charts/charts.less (hunk 3 above)
Comment (dgutride) [inline src/charts/charts.less]: Not yet - we'll make sure one gets in

File: src/charts/topology/examples/topology-view.js (hunk 6 above)
Comment (dgutride) [inline src/charts/topology/examples/topology-view.js]: Yes, due to the fact that they are text icons - this is the way they'll need to be styled.

**Pull request merged**